internal/csi-addons/rbd/reclaimspace.go

```diff
@@ -37,12 +37,13 @@ import (
 // of CSI-addons reclaimspace controller service spec.
 type ReclaimSpaceControllerServer struct {
 	*rs.UnimplementedReclaimSpaceControllerServer
+	*util.VolumeLocks
```

Comment:
that is not exactly what I meant, please consider this, making it a private/internal member of the struct:

```go
type ReclaimSpaceControllerServer struct {
    *rs.UnimplementedReclaimSpaceControllerServer

    // volumeLocks are shared between the CSI Controller
    volumeLocks *util.VolumeLocks
}

func NewReclaimSpaceControllerServer(vl *util.VolumeLocks) *ReclaimSpaceControllerServer {
    return &ReclaimSpaceControllerServer{
        volumeLocks: vl,
    }
}
```

That way, the same instance of VolumeLocks can be shared between both of the controllers.

Reply:
@nixpanic, based on your comment https://github.com/ceph/ceph-csi/issues/4648#issuecomment-2140176480, If there is no downside of sharing VolumeLocks? then would it be safe to share VolumeLocks with rbd ControllerServer?

Something like this?
```go
return &ReclaimSpaceControllerServer{
    volumeLocks: rbdControllServer.VolumeLocks,
}
```